File: CS/16_Print/PrintPdfDocument.cs
using Spire.Pdf;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;

namespace PrintPdfDocument
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e)
        {
            // Create a new PDF document
            PdfDocument doc = new PdfDocument();

            // Load an existing PDF file into the document
            doc.LoadFromFile(@"..\..\..\..\..\..\Data\PrintPdfDocument.pdf");

            // Create a print dialog to select the printer and pages to print
            PrintDialog dialogPrint = new PrintDialog();
            dialogPrint.AllowPrintToFile = true;
            dialogPrint.AllowSomePages = true;
            dialogPrint.PrinterSettings.FromPage = 1;
            dialogPrint.PrinterSettings.ToPage = doc.Pages.Count;

            // If the user clicks OK in the print dialog, proceed with printing
            if (dialogPrint.ShowDialog() == DialogResult.OK)
            {
                // Configure the print settings based on the selected printer and page range
                doc.PrintSettings.SelectPageRange(dialogPrint.PrinterSettings.FromPage, dialogPrint.PrinterSettings.ToPage);
                doc.PrintSettings.PrinterName = dialogPrint.PrinterSettings.PrinterName;

                // Print the document using the specified settings
                doc.Print();
            }
        }
    }
}

File: VB/16_Print/PrintPdfDocument.vb
Imports Spire.Pdf

Namespace PrintPdfDocument
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub button1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button1.Click
			' Create a new instance of PdfDocument object
			Dim doc As New PdfDocument()

			' Load the PDF document from the specified file path
			doc.LoadFromFile("..\..\..\..\..\..\Data\PrintPdfDocument.pdf")

			' Create a new instance of PrintDialog object
			Dim dialogPrint As New PrintDialog()

			' Allow printing to a file
			dialogPrint.AllowPrintToFile = True

			' Allow selection of specific pages for printing
			dialogPrint.AllowSomePages = True

			' Set the starting page for printing to the first page
			dialogPrint.PrinterSettings.FromPage = 1

			' Set the ending page for printing to the total number of pages in the document
			dialogPrint.PrinterSettings.ToPage = doc.Pages.Count

			' If the print dialog is shown and the user clicks "OK"
			If dialogPrint.ShowDialog() = DialogResult.OK Then

				' Select the page range based on the settings from the print dialog
				doc.PrintSettings.SelectPageRange(dialogPrint.PrinterSettings.FromPage, dialogPrint.PrinterSettings.ToPage)

				' Set the printer name for printing to the selected printer from the print dialog
				doc.PrintSettings.PrinterName = dialogPrint.PrinterSettings.PrinterName

				' Print the document using the configured settings
				doc.Print()
			End If

			' Close the document after printing
			doc.Close()
		End Sub
	End Class
End Namespace
